src/backend/distributed/commands/alter_table.c

```diff
@@ -1315,6 +1346,8 @@ ErrorIfUnsupportedCascadeObjects(Oid relationId)
 	bool unsupportedObjectInDepGraph =
 		DoesCascadeDropUnsupportedObject(RelationRelationId, relationId, nodeMap);
 
+	hash_delete_all(nodeMap);
```

Comment:
unrelated change. Why do we need this?

Reply:
This one and also foreign key related one was left over from my local tests. They are unrelated to leakage so I removed them.